lab-davidw/test/test-server.js

```diff
@@ -0,0 +1,100 @@
+
+'use strict';
+
+const server = require('../server');
+const chai = require('chai');
+const http = require('chai-http');
+const expect = chai.expect;
+const app = ('express');
+
+chai.use(http);
+// app.use(router);
+
+describe('server module', function() {
+  before(done => {
+    app.listen(3000);
+    done();
+  });
+  after(done => {
+    server.close();
+    done();
+  });
+});
+
+describe('GET method', function() {
+  let resource;
+  before(done => {
+    chai.request(server)
+    .post('/api/lure')
+    .send({name: 'momba', type: 'rattler', targets: 'trout'})
+    .end((err, res) => {
+      resource = JSON.parse(res.text.toString());
+      done();
+    });
+  });
+  after(done => {
+    chai.request(server)
+    .delete('/api/lure')
+    .query({id: resource.id})
+    .end((err, res) => {
+      console.log(res);
+      done();
+    });
+  });
+
+  describe('api/lure route', function() {
+    describe('a properly formatted request', function() {
+      it('should return a resource given proper id', done => {
+        chai.request(server)
+        .get('/api/lure?id=${resource.id}')
```

Comment:
So the way you have your routes set up now, you actually don't need the ?, I would look how you wrote your endpoint and see the difference here. I believe that is the reason why some of these tests are not passing.